test/cmdlineargs.jl

```diff
@@ -139,7 +139,7 @@ let exename = `$(Base.julia_cmd()) --startup-file=no --color=no`
     # handling of @projectname in --project and JULIA_PROJECT
     let expanded = abspath(Base.load_path_expand("@foo"))
         @test expanded == readchomp(`$exename --project='@foo' -e 'println(Base.active_project())'`)
-        @test expanded == readchomp(addenv(`$exename -e 'println(Base.active_project())'`, "JULIA_PROJECT" => "@foo", "HOME" => homedir()))
+        @test expanded == readchomp(addenv(`$exename -e 'println(Base.active_project())'`, "JULIA_PROJECT" => "@foo", "HOME" => homedir(), "JULIA_DEPOT_PATH" => "$(Base.DEPOT_PATH[1])"))
```

Comment:
```suggestion
        env = copy(ENV)
        env["JULIA_PROJECT"] = "@foo"
        env["HOME"] = homedir()
        delete!(env, "JULIA_DEPOT_PATH")
        @test expanded == readchomp(addenv(`$exename -e 'println(Base.active_project())'`, env))
```

Reply:
@ngam Try this out locally.

Reply:
✅

Reply:
I like your way of doing this because it makes it much clearer, as opposed to the previous implementation.

Reply:
Let me think a little about this and I will revisit. We are trying to figure out a decent solution for the conda-forge's build of julia (basically conda a whole package manager, and julia has effectively another package manager as well at this point, and so we want to make sure we isolate them carefully so that people can get the full use of julia from conda-forge)

Reply:
I think that explicitly deleting the `JULIA_DEPOT_PATH` environment variable (by doing `delete!(env, "JULIA_DEPOT_PATH")`) is the most correct solution here, because it makes explicit what we are trying to accomplish.